File: 1063-best-sightseeing-pair/1063-best-sightseeing-pair.java
class Solution {
    public int maxScoreSightseeingPair (int [] values) {
        
        int ans = 0;
        int prev = values [0];

        for (int i = 1; i < values.length; i++) {
            ans = Math.max (ans, prev + values [i] - i);
            prev = Math.max (prev, values [i] + i);
        }

        return ans;
    }
}

File: 1063-best-sightseeing-pair/1063-best-sightseeing-pair.c
int max (int a, int b) { return a > b ? a : b; }

int maxScoreSightseeingPair (int* values, int valuesSize) {

    int ans= 0;
    int prev = values [0];

    for (int j = 1; j < valuesSize; j++) {
        ans = max (ans, prev + values [j] - j);
        prev = max (prev, values [j] + j);
    }

    return ans;
}